From 0f9e801c04f31b0b173ec22725d08122ecbfaea6 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@localhost.localdomain>
Date: Wed, 3 Oct 2018 09:00:40 -0400
Subject: [PATCH] kms: call LeaveVT on shutdown

It's important that LeaveVT is called on shutdown, since
the X server will leave the current vt on shutdown.

This commit adds the required call, and fixes some confusion
when one X server is starting at around the same time another
X server is shutting down.
---
 src/qxl_kms.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/qxl_kms.c b/src/qxl_kms.c
index 47d7a34..b599908 100644
--- a/src/qxl_kms.c
+++ b/src/qxl_kms.c
@@ -132,6 +132,9 @@ qxl_close_screen_kms (CLOSE_SCREEN_ARGS_DECL)
 
     result = pScreen->CloseScreen (CLOSE_SCREEN_ARGS);
 
+    if (pScrn->vtSema)
+	pScrn->LeaveVT (VT_FUNC_ARGS);
+
     return result;
 }
 
-- 
2.24.1

